Dockerfile-proxy

```diff
@@ -19,6 +19,12 @@ ARG TARGETARCH
 RUN (proxy=$(bin/fetch-proxy $(cat proxy-version) $TARGETARCH) && \
     mv "$proxy" linkerd2-proxy)
 
+FROM --platform=$BUILDPLATFORM docker.io/curlimages/curl:7.75.0  as await
```

Comment:
Would it not be better to reuse the existing `fetch` stage though since that is all this one does? This reduces the need to download a separate image for the singular purpose of fetching the linkerd-await binary.

Reply:
Yep makes sense!